cabot/metricsapp/tasks.py

```diff
@@ -83,6 +83,21 @@ def sync_all_grafana_checks(validate_sites=True):
         sync_grafana_check.apply_async(args=(check.id, str(sync_time)))
 
 
+def grafana_query_diff(old_queries, new_queries):
+    # type: (dict, dict) -> str
+    """returns human-readable diff between two Grafana query strings"""
+
+    # dump=True converts the result to a string and replaces the special keys
+    # jsondiff.delete and jsondiff.insert with the strings "$delete" and "$insert"
+    # if we don't do dump=True, json.dumps will sometimes barf because it doesn't know what to do with these symbols
+    diff = jsondiff.diff(old_queries, new_queries, dump=True)
```

Comment:
Looks like you can pass in your own "dumper" with the args you want (see [1](https://github.com/ZoomerAnalytics/jsondiff/blob/master/jsondiff/__init__.py#L364), [2](https://github.com/ZoomerAnalytics/jsondiff/blob/master/jsondiff/__init__.py#L35)).

Example:
```
>>> a = {'hi': 1, 'bye': 2, 'ok': 3}
>>> b = {'hi': 1, 'bye': 4, 'ok': 6}
>>> c = json.dumps(a)
>>> d = json.dumps(b)
>>> jsondiff.diff(a, b, dump=True, dumper=JsonDumper(sort_keys=True, indent=2))
'{\n  "bye": 4, \n  "ok": 6\n}'
```
Seems better/less hacky than doing the work multiple times

Reply:
This is a much better way, thanks!